test/extract/transpile/svelte-wrap.spec.js

```diff
@@ -0,0 +1,26 @@
+const { promises: fsPromise } = require("fs");
+const { expect } = require("chai");
+const normalizeNewline = require("normalize-newline");
+const wrap = require("../../../src/extract/transpile/svelte-wrap")(
+  require("../../../src/extract/transpile/typescript-wrap")(true)
+);
+
+describe("svelte transpiler", () => {
+  it("tells the svelte transpiler is available", () => {
+    expect(wrap.isAvailable()).to.equal(true);
+  });
+
+  it("'transpiles' svelte", async () => {
+    const observedPromise = fsPromise
+      .readFile("./test/extract/transpile/fixtures/svelte.svelte", "utf8")
```

Comment:
Well, it's the same regardless of whether it's browser or server, https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/ . It's just faster, and as I understood, this nodejs process also yields CPU resources to other processes when it's doing async IO task and no microtask left in eventloop (while with sync IO task, it will hold on to CPU resources, but do not utilize it)

Reply:
No new reasons I wasn't aware of :-) It's fine to keep using async code, but I'm always curious for new reasons to actually start using it in a command line tool.

It seems the nice and elegant fs.promises is not supported on node 10 until 10.17, though and I'd rather not major-bump over functionality that is only used in unit tests...

> The performance advantage is there for concurrent programs, where other javascript can be executed while
> during the execution of i/o. In these cases async is actually the only way to have decent functionality.

Reply:
Turn our, `fs.promise` is already on 10 https://nodejs.org/en/blog/release/v10.0.0/  https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fspromises_readfile_path_options

Reply:
You're right! Wonder where that linter warning about 10.17 came from